dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/ESClient.java

```diff
@@ -312,59 +321,35 @@ private void restartNode(Builder externalSettings) {
         initNode(externalSettings);
     }
 
+    private String getServerAddress(final Server server) {
+        return  (UtilMethods.isSet(server.getHost()) && !server.getHost().equals("localhost"))
+            ? server.getHost()
+            : server.getIpAddress();
+    }
+
     /**
      *
-     * @param serverAPI
-     * @param currentServer
      * @param externalSettings
      * @throws DotDataException
      */
-    private void setUnicastHosts(final ServerAPI serverAPI,
-            Server currentServer, final Builder externalSettings) throws DotDataException {
-
-        String initData;
+    private void setUnicastHosts(final Builder externalSettings) throws DotDataException {
 
         final String bindAddr = externalSettings.get("transport.host");
-        final String transportTCPPort = externalSettings
-                .get(ServerPort.ES_TRANSPORT_TCP_PORT.getPropertyName());
-
-        final List<Server> aliveServers = serverAPI
-                .getAliveServers(Collections.singletonList(currentServer.getServerId()));
-
-        currentServer = Server.builder(currentServer)
-                .withEsTransportTcpPort(Integer.parseInt(transportTCPPort)).build();
-        aliveServers.add(currentServer);
+        final String transportTCPPort = externalSettings.get(ServerPort.ES_TRANSPORT_TCP_PORT.getPropertyName());
 
-        final StringBuilder initialHosts = new StringBuilder();
+        final List<Server> aliveServers = serverAPI.getAliveServers();
```

Comment:
Since we already updated our current server with the propert ports, we just can get them all and iterate to form our initialHosts lists, instead of excluding myself from the list and them add me back.